server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -533,15 +537,22 @@ private static Map<String, Mapper> buildMergedMappers(
             MergeReason reason,
             MapperMergeContext objectMergeContext
         ) {
-            Map<String, Mapper> mergedMappers = null;
+            Map<String, Mapper> mergedMappers = new HashMap<>(existing.mappers);
             for (Mapper mergeWithMapper : mergeWith) {
-                Mapper mergeIntoMapper = (mergedMappers == null ? existing.mappers : mergedMappers).get(mergeWithMapper.simpleName());
```

Comment:
I believe that we are attempting not to copy the map into a new one here. See #83776 . I am not sure if this is necessary, but I would leave it as-is unless if possible. oh well actually down below we do the same copy. That's odd. If you were cleaning things up, could we discuss those changes separately unless they are necessary to make your change work?

Reply:
I've extracted some of the mechanical changes in this PR: https://github.com/elastic/elasticsearch/pull/104578

Reply:
Looks much cleaner now that that PR has been merged.